nixos/modules/services/games/minecraft-server.nix

```diff
@@ -210,6 +242,25 @@ in {
           rm .declarative
         fi
       '');
+
+      preStop = let 
+        script = pkgs.writeShellScript "minecraft-stop-commands" cfg.stopCommands;
+      in optionalString (cfg.stopCommands != null) ''
+        ${script} > ${cfg.standardInputFifo}
+      '';
+    };
+
+    systemd.sockets.minecraft-server = {
+      description = "Minecraft stdin fifo file";
+      socketConfig = {
+        ListenFIFO = cfg.standardInputFifo;
+        SocketUser = "minecraft";
+        SocketGroup = "minecraft";
+        SocketMode = "0660";
+        DirectoryMode = "0750";
```

Comment:
Seems like the root user owns the directory, meaning this makes other users unable to access it. The default of `0755` should work, so this can be removed.